backend/backend.go

```diff
@@ -110,3 +110,58 @@ func UnmarshalHistoryEvent(bytes []byte) (*HistoryEvent, error) {
 	}
 	return e, nil
 }
+
+func GetSubOrchestrationInstances(ctx context.Context, be Backend, iid api.InstanceID, includeAll bool) ([]api.InstanceID, error) {
+	owi := &OrchestrationWorkItem{
+		InstanceID: iid,
+	}
+	state, err := be.GetOrchestrationRuntimeState(ctx, owi)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch orchestration state for '%s': %w", iid, err)
+	}
+	oldEvents := state.OldEvents()
+	newEvents := state.NewEvents()
+
+	subOrchestrationInstancesTotal := make([]api.InstanceID, 0)
+	subOrchestrationMap := make(map[int32]api.InstanceID)
```

Comment:
Please allocate with an initial capacity - ok to be worst-case scenario

Reply:
Done